src/Exception/NestedTspanException.php

```diff
@@ -4,40 +4,12 @@
 namespace App\Exception;
 
 use DOMElement;
-use Exception;
 
-class NestedTspanException extends Exception
+class NestedTspanException extends SvgStructureException
 {
 
-    /** @var DOMElement */
-    protected $tspan;
-
     public function __construct(DOMElement $tspan)
     {
-        $this->tspan = $tspan;
-        $msg = 'Nested tspan elements are not supported '
-            .'(ID: "'.$this->getTspanId().'"; text: "'.$this->getTextContent().'")';
-        parent::__construct($msg);
-    }
-
-    /**
-     * Get the closest ID to the nested tspan.
-     * @return string
-     */
-    public function getTspanId(): string
-    {
-        $el = $this->tspan;
-        $id = '';
-        // Traverse up the tree to find an ID.
-        while (!$id && 'svg' !== $el->nodeName) {
-            $id = $el->getAttribute('id');
-            $el = $el->parentNode;
-        }
-        return $id;
-    }
-
-    public function getTextContent(): string
-    {
-        return $this->tspan->textContent;
+        parent::__construct('Nested tspan elements are not supported', $tspan);
```

Comment:
Does it matter that the parent is expecting a `DOMNode` rather than a `DOMElement`?

Reply:
DOMElement is a subclass of DOMNode, so it should be fine.